File: admin/api/tag.ts
import api from "@/lib/axios";
import { TagResponse } from "@/types/tag";

export async function createTag(data: any) {
    const response = await api.post(
        "/admin/tags",
        data
    );

    return response.data;
}

export async function getTags(
    page = 1,
    limit = 20,
    search = ""
) {
    const response = await api.get<TagResponse>(
        `/tags?page=${page}&limit=${limit}&search=${search}`
    );

    return response.data;
}

export async function updateTag(data: any) {
    const response = await api.put(
        `/admin/tags/${data.id}`,
        data
    );

    return response.data;
}

export async function deleteTag(id: number) {
    const response = await api.delete(
        `/admin/tags/${id}`
    );

    return response.data;
}

File: backend/routes/tag.go
package routes

import (
	"github.com/gin-gonic/gin"

	"github.com/techwebcode/techwebcode/backend/controller"
)

// Public Routes
func RegisterTagPublicRoutes(router *gin.RouterGroup, controller *controller.TagController) {
	router.GET("/tags", controller.GetTags)
	router.GET("/Tags", controller.GetTags)
}

// Admin Routes
func RegisterTagAdminRoutes(router *gin.RouterGroup, controller *controller.TagController) {

	router.POST("/tags", controller.CreateTag)
	router.POST("/Tags", controller.CreateTag)

	router.GET("/tags", controller.GetTags)
	router.GET("/Tags", controller.GetTags)

	router.PUT("/tags/:id", controller.UpdateTag)
	router.PUT("/Tags/:id", controller.UpdateTag)

	router.DELETE("/tags/:id", controller.DeleteTag)
	router.DELETE("/Tags/:id", controller.DeleteTag)
}